logos-codegen/src/graph/mod.rs

```diff
@@ -58,9 +58,18 @@ pub trait Disambiguate {
 
 /// Id of a Node in the graph. `NodeId` can be referencing an empty
 /// slot that is going to be populated later in time.
-#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
 pub struct NodeId(NonZeroU32);
 
+impl Hash for NodeId {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        // Always use little-endian byte order for hashing to avoid
+        // different code generation on big-endian platforms due to
+        // iteration over a HashMap
+        state.write(&self.0.get().to_le_bytes())
```

Comment:
```suggestion
    fn hash<H: Hasher>(&self, state: &mut H) {
        // Always use little-endian byte order for hashing to avoid
        // different code generation on big-endian platforms due to
        // iteration over a HashMap,
        // see https://github.com/maciejhirsz/logos/issues/427.
        state.write(&self.0.get().to_le_bytes())
```

For documentation purposes :-)